pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -628,6 +623,55 @@ func upgradeEligibilityCheck(upgrade *harvesterv1.Upgrade, repoInfo *RepoInfo) (
 	return true, ""
 }
 
+func (h *upgradeHandler) prepareNodesForUpgrade(upgrade *harvesterv1.Upgrade, repoInfoStr string) (*harvesterv1.Upgrade, error) {
+	upgradeConfig, err := settings.DecodeConfig[settings.UpgradeConfig](settings.UpgradeConfigSet.Get())
+	if err != nil {
+		return upgrade, err
+	}
+	logrus.WithFields(logrus.Fields{
+		"namespace":      upgrade.Namespace,
+		"name":           upgrade.Name,
+		"upgrade_config": upgradeConfig,
+	}).Info("start preparing nodes for upgrade")
+
+	nodes, err := h.nodeCache.List(labels.Everything())
+	if err != nil {
+		return upgrade, err
+	}
+
+	imagePreloadConcurrency := upgradeConfig.PreloadOption.Strategy.Concurrency
+	switch upgradeConfig.PreloadOption.Strategy.Type {
```

Comment:
Correct me if I'm wrong, looks like we didn't handle that `sequential` Strategy Type here, and the imagePreloadConcurrency is 0 (default concurrency value).  And that makes `sequential` the same behavior as `Parallel` (preload to all node concurrently)

Reply:
Good catch. I missed that part during refactoring, meaning it doesn't have unit tests covered. I'll fix that.